watershed-android/watershed/app/src/main/java/com/blueprint/watershed/Tasks/TaskFragment.java

```diff
@@ -113,23 +104,43 @@ public void onRefresh() {
                 getTasksRequest();
             }
         });
-    }
 
-    /**
-     * Starts a new fragment for the corresponding task view
-     * @param l - ListView that was clicked
-     * @param v - View that was clicked
-     * @param position - Position of view
-     * @param id = id of View
-     */
-    @Override
-    public void onListItemClick(ListView l, View v, int position, long id) {
-        Task taskClicked;
-        if (TASK_TYPE == USER) taskClicked = mUserTaskList.get(position);
-        else taskClicked = mAllTaskList.get(position);
+        mListView = (ExpandableListView) view.findViewById(android.R.id.list);
+        mListView.setOnChildClickListener(new ExpandableListView.OnChildClickListener() {
+            @Override
+            public boolean onChildClick(ExpandableListView expandableListView, View view, int groupPosition, int childPosition, long l) {
+                Log.i("asdf", "clicked");
+                Task taskClicked;
+                if (mArgs.getInt(OPTION) == USER) taskClicked = mUserTaskList.get(mTaskListHeaders.get(groupPosition)).get(childPosition);
+                else taskClicked = mAllTaskList.get(mTaskListHeaders.get(groupPosition)).get(childPosition);
+
+                TaskDetailFragment detailFragment = TaskDetailFragment.newInstance(taskClicked);
+                mParentActivity.replaceFragment(detailFragment);
+                return false;
+            }
+        });
+        mListView.setOnScrollListener(new AbsListView.OnScrollListener() {
+            @Override
+            public void onScrollStateChanged(AbsListView view, int scrollState) {}
+
+            @Override
+            public void onScroll(AbsListView view, int firstVisibleItem, int visibleItemCount, int totalItemCount) {
+                int topRowVerticalPosition = (mListView == null || mListView.getChildCount() == 0) ?
+                        0 : mListView.getChildAt(0).getTop();
+                mSwipeLayout.setEnabled((topRowVerticalPosition >= 0));
+            }
+        });
+        mNoTasks = (RelativeLayout) view.findViewById(R.id.no_tasks_layout);
 
-        TaskDetailFragment detailFragment = TaskDetailFragment.newInstance(taskClicked);
-        mParentActivity.replaceFragment(detailFragment);
+        if (mArgs.getInt(OPTION) == USER) {
+            mUserTaskAdapter = new TaskAdapter(mParentActivity, mTaskListHeaders, mUserTaskList);
+            mListView.setAdapter(mUserTaskAdapter);
+        }
+        else {
+            mAllTaskAdapter = new TaskAdapter(mParentActivity, mTaskListHeaders, mAllTaskList);
+            mListView.setAdapter(mAllTaskAdapter);
+        }
+//        mListView.setEmptyView(mNoTasks);
```

Comment:
If we're not using this, we shouldn't have it here. Also, if this is commented out, then the above

<code>mNoTasks = (RelativeLayout) view.findViewById(R.id.no_tasks_layout);</code>

is not needed anymore.